airflow/decorators/__init__.py

```diff
@@ -15,34 +15,58 @@
 # specific language governing permissions and limitations
 # under the License.
 
-from typing import Callable, Optional
-
-from airflow.decorators.python import python_task
-from airflow.decorators.python_virtualenv import _virtualenv_task
+from airflow.decorators.python import PythonDecoratorMixin
+from airflow.decorators.python_virtualenv import PythonVirtualenvDecoratorMixin
 from airflow.decorators.task_group import task_group  # noqa
 from airflow.exceptions import AirflowException
 from airflow.models.dag import dag  # noqa
 from airflow.providers_manager import ProvidersManager
 
+# [START import_docker]
+try:
+    from airflow.providers.docker.decorators.docker import DockerDecoratorMixin
+except ImportError:
+    DockerDecoratorMixin = None
+# [END import_docker]
+
 
 class _TaskDecorator:
```

Comment:
We can mix these ones in directly:

```python
class _TaskDecorator(PythonDecoratorMixin, DockerDecoratorMixin):
```

Reply:
@ashb I don't think you can import from None

```
Python 3.7.8 (default, Oct  8 2020, 09:28:20)
[Clang 12.0.0 (clang-1200.0.32.2)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> class Foo(None):
...     def bar():
...             pass
...
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: NoneType takes no arguments
>>>
```

Reply:
Typo, I meant to say

```python
class _TaskDecorator(PythonDecoratorMixin, PythonVirtualenvDecoratorMixin)
```